brave/src/main/java/brave/Tracing.java

```diff
@@ -0,0 +1,215 @@
+package brave;
+
+import brave.internal.Internal;
+import brave.internal.Nullable;
+import brave.internal.Platform;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import brave.sampler.Sampler;
+import java.io.Closeable;
+import zipkin.Endpoint;
+import zipkin.reporter.AsyncReporter;
+import zipkin.reporter.Reporter;
+import zipkin.reporter.Sender;
+
+/**
+ * This provides utilities needed for trace instrumentation. For example, a {@link Tracer}.
+ *
+ * <p>Instances built via {@link #newBuilder()} are registered automatically such that statically
+ * configured instrumentation like JDBC drivers can use {@link #current()}.
+ *
+ * <p>This type can be extended so that the object graph can be built differently or overridden, for
+ * example via spring or when mocking.
+ */
+public abstract class Tracing implements Closeable {
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  abstract public Tracer tracer();
+
+  abstract public Propagation.Factory propagationFactory();
+
+  // Implementations should override to cache this.
+  public Propagation<String> stringPropagation() {
+    return propagationFactory().create(Propagation.KeyFactory.STRING);
+  }
+
+  /** Used internally by operations such as {@link Span#finish()}, exposed for convenience. */
+  abstract public Clock clock();
+
+  // volatile for visibility on get. writes guarded by Tracing.class
+  static volatile Tracing current = null;
+
+  /**
+   * Returns the most recently created tracer iff it hasn't been closed. null otherwise.
+   *
+   * <p>This object should not be cached.
+   */
+  @Nullable public static Tracing current() {
+    return current;
+  }
+
+  /** Ensures this component can be garbage collected, by making it not {@link #current()} */
+  @Override abstract public void close();
+
+  public static final class Builder {
+    String localServiceName;
+    Endpoint localEndpoint;
+    Reporter<zipkin.Span> reporter;
+    Clock clock;
+    Sampler sampler = Sampler.ALWAYS_SAMPLE;
```

Comment:
I think we have done this mainly a preference for first starters.. I think finagle defaults to 0.1, which is good but first starters tend to feel things are broke (fwiw this isn't a change I know it is hard to see that)